File: tfdma/src/main/java/org/openchai/tensorflow/api/Logger.java
package org.openchai.tensorflow.api;

import java.io.PrintWriter;
import java.io.StringWriter;
import java.util.Date;

public class Logger {

  public static void p(String msg) {
    info(msg);
  }

  public static void p(String msg, Object arg1, Object ... moreArgs) {

    info(f(msg,arg1, moreArgs));
  }

  public static String f(String msg, Object ... v) {
    return String.format(msg, v);
  }

  public static void debug(String msg) {
    System.out.println(iformat("Debug: " + msg));
  }

  public static void info(String msg) {
    System.out.println(iformat("Info: " + msg));
  }

  public static void warn(String msg, Exception e) {
    System.err.println(String.format("WARN: %s: %s", msg, eToString(e)));
  }

  public static void error(String msg, Exception e) {
    System.err.println(String.format("ERROR: %s: %s", msg, eToString(e)));
  }

  static String iformat(String msg) {
    Date d = new java.util.Date();
    String dm = String.format("%02d%02d %02d:%02d:%02d.%03d",
            d.getMonth(), d.getDate(), d.getHours(), d.getMinutes(),
            d.getSeconds(), d.getTime() / new Double(10e10).intValue());
    return String.format("[%s] %s",dm, msg);
  }

  public static String eToString(Exception e) {
    StringWriter sw = new StringWriter();
    PrintWriter pw = new PrintWriter(sw);
    e.printStackTrace(pw);
    return String.format("%s - %s", e.getMessage(), sw.toString());
  }
}

File: tcpclient/src/main/scala/org/openchai/tcp/util/Logger.scala
package org.openchai.tcp.util

import java.io.{PrintWriter, StringWriter}

trait Logger {
  def f(msg: String) = {
    val d = new java.util.Date
//    "%02d%02d-%02d:%02d:%02d.%03d".format(d.getMonth, d.getDate, d.getHours, d.getMinutes, d.getSeconds, d.getTime / 10e10.toInt)
    "%02d:%02d.%03d".format(d.getHours, d.getMinutes, d.getSeconds, d.getTime / 10e10.toInt)
  }

  def debug(msg: String) = Logger.debug(msg)
  def info(msg: String) = Logger.info(msg)
  def warn(msg: String) = Logger.warn(msg)
  def error(msg: String) = Logger.error(msg)
}
object Logger {

  val LogLevel = Option(System.getProperty("logger.level")).getOrElse("2").toInt
  def f(msg: String) = {
    val d = new java.util.Date
    "[%02d%02d-%02d:%02d:%02d.%03d] %s".format(d.getMonth, d.getDate, d.getHours, d.getMinutes, d.getSeconds, d.getTime / 10e10.toInt, msg)

  }
  def f2(msg: String) = {
    val d = new java.util.Date
    "[%02d:%02d:%02d.%02d] %s".format(d.getHours, d.getMinutes, d.getSeconds, d.getTime / 10e11.toInt, msg)
  }

  def debug(msg: Any) = if (LogLevel >= 3) println(f2(s"Debug: $msg"))
  def info(msg: Any) = if (LogLevel >= 2) println(f2(s"Info: $msg"))
  def gpuinfo(msg: Any) = if (LogLevel >= 1) println(f2(s"GPU: $msg"))
  def warn(msg: Any) = if (LogLevel >= 1) println(f2(s"WARN: $msg"))
  def error(msg: Any) = if (LogLevel >= 0) println(f2(s"ERROR: $msg"))
  def error(msg: Any, t: Throwable) = if (LogLevel >= 0) println(f2(s"ERROR: $msg\n${toString(t)}"))

  def toString(t: Throwable) = {
    val sw = new StringWriter()
    val pw = new PrintWriter(sw)
    t.printStackTrace(pw)
    s"$t ${t.getMessage} - ${sw.toString}"
  }

}
